test/scenarios/fork/StakingNodeProofSubmission-Scenario.spec.sol

```diff
@@ -0,0 +1,301 @@
+// SPDX-License-Identifier: BSD 3-Clause License
+pragma solidity ^0.8.24;
+import {StakingNodesManager} from "src/StakingNodesManager.sol";
+import {OwnableUpgradeable} from "lib/openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import {ynETH} from "src/ynETH.sol";
+import {RewardsReceiver} from "src/RewardsReceiver.sol";
+import {RewardsDistributor} from "src/RewardsDistributor.sol";
+import {ProxyAdmin} from "lib/openzeppelin-contracts/contracts/proxy/transparent/ProxyAdmin.sol";
+import {IRewardsDistributor} from "src/interfaces/IRewardsDistributor.sol";
+import {IStakingNodesManager} from "src/interfaces/IStakingNodesManager.sol";
+import {IStakingNode} from "src/interfaces/IStakingNodesManager.sol";
+import {IBeaconChainOracle} from "lib/eigenlayer-contracts/src/contracts/interfaces/IBeaconChainOracle.sol";
+import {IStrategy} from "lib/eigenlayer-contracts/src/contracts/interfaces/IStrategy.sol";
+import {TransparentUpgradeableProxy} from "lib/openzeppelin-contracts/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {ITransparentUpgradeableProxy} from "lib/openzeppelin-contracts/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import {IERC20} from "lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {ScenarioBaseTest} from "test/scenarios/ScenarioBaseTest.sol";
+import { Invariants } from "test/scenarios/Invariants.sol";
+import {stdStorage, StdStorage} from "forge-std/Test.sol"; 
+import {BytesLib} from "lib/eigenlayer-contracts/src/contracts/libraries/BytesLib.sol";
+import { MockEigenLayerBeaconOracle } from "test/mocks/MockEigenLayerBeaconOracle.sol";
+
+import {UpgradeableBeacon} from "lib/openzeppelin-contracts/contracts/proxy/beacon/UpgradeableBeacon.sol";
+import {TestStakingNodesManagerV2} from "test/mocks/TestStakingNodesManagerV2.sol";
+import {TestStakingNodeV2} from "test/mocks/TestStakingNodeV2.sol";
+
+import {BeaconChainProofs} from "lib/eigenlayer-contracts/src/contracts/libraries/BeaconChainProofs.sol";
+import {Merkle} from "lib/eigenlayer-contracts/src/contracts/libraries/Merkle.sol";
+import { ProofParsingV1 } from "test/eigenlayer-utils/ProofParsingV1.sol";
+import {Utils} from "script/Utils.sol";
+import "forge-std/console.sol";
+
+
+contract StakingNodeTestBase is ScenarioBaseTest, ProofParsingV1 {
```

Comment:
we have a more general problem with tests:



Tests that execute all tests on a fresh deployment vs tests that assume the deployment is the existing deployment.



The latter is the most accurate, since we roll the upgrade for ynETH on top of the existing deployment.


"Fresh" integration tests on a fresh deployment have the advantage that we can push the system to any configuration, while the mainnet one keeps moving underneat you (may break tests randomly, won't be able to reproduce certain system states without lots of prank-ing and etching which introduce error-prone additional mutations)